src/Components/AddComment.jsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { useEffect, useState, useContext } from "react";
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import { UserContext } from "../../context/userContext";
+import { postNewComment } from "../api";
+import { useParams } from "react-router-dom";
+import CommentCard from "./CommentCard";
+
+
+function AddComment () {
+    const {id} = useParams();
+    const { user:{username, name, avatar_url} } = useContext(UserContext);
+    const [commentBody, setCommentBody] = useState("");
+    const [placholderBody, setPlaceholderBody] = useState("Insert comment...")
+    const [message, setMessage] = useState("Post a Comment")
+
+    function handleClick (event) {
+        event.preventDefault()
+
+        if (commentBody !== "") {
+          const input = { username: username, body: commentBody }
+          postNewComment(id, input)
+          .then((result) => {
+            setMessage("Comment loading")
+            CommentCard(result).then(() => {
+              console.log(result);
+              setMessage("Comment uploaded!")
+
+            })
```

Comment:
Currently the comment only loads on refresh, I get the message that the comment is loading but it never comes. 

Before you use postNewComment, if you update the full comments state and add in this new comment the user will see the comment load up immediately. Whenever state is changed it causes a refresh. Doing this would be optimistic rendering 👍